src/operators/generation/mutation.rs

```diff
@@ -652,3 +653,94 @@ mod translocation_mutation {
         );
     }
 }
+
+/// Performs the special Differential Evolution mutation, similar to an arithmetic crossover.
+///
+/// Requires a DE selection directly beforehand, e.g., [DEBest].
+#[derive(Serialize, Deserialize, Clone)]
+pub struct DEMutation {
+    // Number of difference vectors ∈ {1, 2}.
+    y: usize,
+    // Difference vector scaling ∈ (0, 2].
+    f: f64,
+}
+impl DEMutation {
+    pub fn new<P: Problem<Encoding = Vec<f64>> + VectorProblem>(
+        y: usize,
+        f: f64,
+    ) -> Box<dyn Component<P>> {
+        assert!((0.0..=2.0).contains(&f));
+        assert!([1, 2].contains(&y));
+        Box::new(Generator(Self { y, f }))
+    }
+}
+
+impl<P> Generation<P> for DEMutation
+where
+    P: Problem<Encoding = Vec<f64>> + VectorProblem,
+{
+    fn generate_population(
+        &self,
+        population: &mut Vec<P::Encoding>,
+        problem: &P,
+        _state: &mut State,
+    ) {
+        assert_eq!(population.len() % (self.y * 2 + 1), 0);
+
+        let chunks = population.chunks_exact_mut(self.y * 2 + 1);
+
+        // Iterate over chunks of size `y * 2 + 1`, with the first element as base
+        for chunk in chunks {
+            // Compiler can't guarantee that pattern always matches
+            if let [base, remainder @ ..] = chunk {
+                let pairs = remainder.chunks_exact(2).map(|chunk| {
+                    // Compiler can't guarantee that pattern always matches
+                    match chunk {
+                        [s1, s2] => (s1, s2),
+                        _ => unreachable!(),
+                    }
+                });
+
+                for (s1, s2) in pairs {
+                    for i in 0..problem.dimension() {
+                        base[i] += self.f * (s1[i] - s2[i]);
```

Comment:
I think it is possible to leave the boundaries of the search space using this mutation. So we should either clamp the values here (as we already did in other mutations) or implement our constraint handling properly.